libs/wingc/src/capture.rs

```diff
@@ -48,141 +52,46 @@ fn collect_captures(capture_list: Vec<Capture>) -> Captures {
 	captures
 }
 
-pub fn scan_for_inflights_in_scope(scope: &Scope, diagnostics: &mut Diagnostics) {
```

Comment:
Ah, I see. I think it's doable, the second traversal is just a little bit more tricky because as implemented because of the environment and statement index passing. Not sure if I can get to it in this PR but it's a good callout.